apollo-composition/src/lib.rs

```diff
@@ -287,18 +287,19 @@ impl From<SubgraphLocation> for BuildMessageLocation {
 
 impl SubgraphLocation {
     fn from_ast(node: SubgraphASTNode) -> Option<Self> {
-        let loc = node.loc?;
         Some(Self {
-            subgraph: node.subgraph.unwrap_or_default(),
-            range: Some(Range {
-                start: LineColumn {
-                    line: loc.start_token.line?,
-                    column: loc.start_token.column?,
-                },
-                end: LineColumn {
-                    line: loc.end_token.line?,
-                    column: loc.end_token.column?,
-                },
+            subgraph: node.subgraph?,
+            range: node.loc.and_then(|node_loc| {
+                Some(Range {
```

Comment:
```suggestion
            range: node.loc.map(|node_loc| {
                Range {
```

Reply:
I think we need `and_then` to handle all the line/column Options below?

Reply:
ah sorry i missed that!

Reply:
That `?` operator in rust is pretty unreadable in my opinion :P